libs/barista-components/slider/src/slider.ts

```diff
@@ -0,0 +1,452 @@
+   */
+  private _captureEvents(): void {
+    if (!this._platform.isBrowser) {
+      // early exit if env is not browser
+      return;
+    }
+
+    const start$ = merge(
+      fromEvent(this._trackWrapper.nativeElement, 'mousedown'),
+      fromEvent(this._trackWrapper.nativeElement, 'touchstart'),
+    );
+
+    const move$ = merge(
+      fromEvent<MouseEvent>(window, 'mousemove').pipe(
+        map(mouseEvent => mouseEvent.clientX),
+      ),
+      fromEvent<TouchEvent>(window, 'touchmove').pipe(
+        map(touchEvent => touchEvent.changedTouches[0].clientX),
+      ),
+    );
+
+    const moveEnd$ = merge(
+      fromEvent(window, 'mouseup'),
+      fromEvent(window, 'touchend'),
+    );
+
+    /** Combining the start, move and moveEnd observers concludes the drag observer. */
+    const drag$ = start$.pipe(
+      switchMap(() =>
+        move$.pipe(
+          distinctUntilChanged(),
+          observeOn(animationFrameScheduler),
+          takeUntil(moveEnd$),
+        ),
+      ),
+      share(),
+    );
+
+    const click$ = fromEvent<MouseEvent>(
+      this._trackWrapper.nativeElement,
+      'click',
+    ).pipe(map(mouseEvent => mouseEvent.clientX));
+
+    // stream from keyboard events
+    const keyDown$ = fromEvent<KeyboardEvent>(
+      this._trackWrapper.nativeElement,
+      'keydown',
+    ).pipe(
+      filter(keyboardEvent => KEY_CODES_ARRAY.includes(keyboardEvent.keyCode)),
+      map(keyboardEvent => {
+        keyboardEvent.stopPropagation(); // global angular keydown would trigger CD.
+        const valueAddition = getKeyCodeValue(
+          this._max,
+          this.step,
+          keyboardEvent.keyCode,
+        );
+        const rounded = roundToDecimal(
+          this._value + valueAddition,
+          this._roundShift,
+        );
+        return clamp(rounded, this._min, this._max);
+      }),
+      distinctUntilChanged(),
+    );
+
+    // triggered by drag and click
+    const mouse$ = merge(drag$, click$).pipe(
+      withLatestFrom(this._clientRect$),
+      map(([coordinate, { left, width }]) => {
+        const newValue = getSliderValueForCoordinate({
+          coordinate,
+          offset: left,
+          width,
+          min: this._min,
+          max: this._max,
+          step: this.step,
+          roundShift: this._roundShift,
+        });
+        return newValue;
+      }),
+      distinctUntilChanged(),
+    );
+
+    const inputValue$ = this._inputFieldValue$.pipe(
+      /**
+       * distinctUntilChanged() purposefully left out, to round the value
+       * and update the input field with the rounded value
+       */
+      map(value => roundToSnap(value, this.step, this._min, this._max)),
+    );
+
+    merge(this._value$, inputValue$, mouse$, keyDown$)
+      .pipe(
+        filter(() => !this._isDisabled),
+        takeUntil(this._destroy$),
+      )
+      .subscribe(value => this._zone.run(() => this._updateSlider(value)));
```

Comment:
Please add braces because the zone run or the update Slider are returning void.
Arrow functions without braces should only be used if you return a value.

```suggestion
      .subscribe(value => {
        this._zone.run(() => {
          this._updateSlider(value);
        })
      });
```

Reply:
It will still return the same I guess, but if it is the convention, I'll add it.